src/lib/components/dynamic-region-form/DynamicForm.svelte

```diff
@@ -12,22 +12,31 @@
 		forEachSubGroup,
 		getFieldErrorMessage
 	} from './utils';
-	import type { RunData } from '$lib/types/userState';
+	import type { EmulatorResults } from '$lib/types/userState';
 
 	interface Props {
 		schema: DynamicFormSchema;
 		initialValues: Record<string, FormValue>;
-		// The first run is user-initiated after entering preRun values, subsequent runs are triggered automatically when fields change - this prop tracks these life cycle stages.
-		hasRun: boolean;
+		hasRunBaseline: boolean;
 		children: Snippet;
-		submit: (formValues: Record<string, unknown>, triggerRun?: boolean) => Promise<RunData | null>;
+		run: (formValues: Record<string, unknown>) => Promise<EmulatorResults | null>;
```

Comment:
```suggestion
		run: (formValues: Record<string, FormValue>) => Promise<EmulatorResults | null>;
```
or is this different?